cdap-docs/developers-manual/source/data-exploration/filesets.rst

```diff
@@ -0,0 +1,86 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _fileset-exploration:
+
+============================================
+Fileset Exploration
+============================================
+
+The ``FileSet`` and ``TimePartitionedFileSet`` datasets can be explored through ad-hoc SQL-like queries.
+To enable exploration, you must set several properties when creating the Dataset, and the files in 
+your dataset must fit certain constraints. These are described in more detail in the following sections. 
+
+Explore Properties
+------------------
+A ``FileSet`` or ``TimePartitionedFileSet`` can be made explorable by setting several properties when
+creating the dataset. For example, in the configure method of your application::
+
+    Schema schema = Schema.recordOf(
+      "purchase",
+      Schema.Field.of("user", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("item_id", Schema.of(Schema.Type.INT)),
+      Schema.Field.of("price", Schema.of(Schema.Type.DOUBLE))
+    );
+    createDataset("my-files", "fileSet", FileSetProperties.builder()
+      .setBasePath("mylocation")
+      .setInputFormat(AvroKeyInputFormat.class)
+      .setOutputFormat(AvroKeyOutputFormat.class)
+      // everything past here is an explore property
+      .setEnableExploreOnCreate(true)
+      .setSerDe("org.apache.hadoop.hive.serde2.avro.AvroSerDe")
+      .setExploreInputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat")
+      .setExploreOutputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat")
+      .setTableProperty("avro.schema.literal", schema.toString())
+      .build());
+
+These Dataset properties map directly to table properties in Hive. 
+In fact, the create table statement that will be generated for the table above is::
```

Comment:
For example, the table above would result in the following "create table" statement being generated::